### PR TITLE
Translate ContextManagerMock to Swift

### PR DESCRIPTION
**Pull request by crazytonyli**
Follow up of https://github.com/wordpress-mobile/WordPress-iOS/pull/18579#discussion_r875734563

This PR rewrites `ContextManagerMock` in Swift. This change itself is quite simple, since there are only a couple functions in `ContextManagerMock`, but the result of removing `ContextManagerMock.h` from the bridging header leads to quite a few Swift files changes - adding `import XCTest`, since the XCTest imports is now removed from the bridging heard file along with `ContextManagerMock.h`.

## Test Instructions

Make sure unit test passes.

## Regression Notes

N/A. No impact on the app.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/WordPressTest/ContextManagerMock.swift
Comment (AliSoftware) [inline WordPress/WordPressTest/ContextManagerMock.swift]: Would be nice to document the use of `/dev/null` for In-Memory model a bit better here (and maybe as part of the `/// parameter storeURL` doc in the `ContextManager` superclass's `init` as well?) 🙃

Comment (crazytonyli) [reply WordPress/WordPressTest/ContextManagerMock.swift]: Yep, I'll add some comments to the initialiser in ContextManager.h

File: WordPress/Classes/Utility/ContextManager.h
Comment (mokagio) [inline WordPress/Classes/Utility/ContextManager.h]: Nitpick: 

```suggestion
 @param storeURL Database location. Use "/dev/null" to create an in-memory database.
```

Reference: [Wikipedia](https://en.wikipedia.org/wiki/In-memory_database) or you can google "in memory database".

I'm sure there's an English grammar rule to know when to hyphen like this, but I don't know it. I've sort of developed an hear for it thanks to Grammarly correcting me for the past 2 years 🙃

File: WordPress/WordPressTest/ContextManagerMock.swift
Comment (mokagio) [reply WordPress/WordPressTest/ContextManagerMock.swift]: A further refinement / nitpick would be to move this URL definition in a constant, to make it clearer what it does. 

Maybe something like:

```swift
super.init(modelName: modelName, store: ContextManager.inMemoryStoreURL)
```

File: WordPress/WordPressTest/ContextManagerMock.swift
Comment (mokagio) [inline WordPress/WordPressTest/ContextManagerMock.swift]: Pedantic / nitpick

```suggestion
        // This log resolves a deadlock we previously experienced in
        // `DashboardCardTests.testShouldNotShowQuickStartIfDefaultSectionIsSiteMenu`
        //
        // Unfortunately, we're not sure why.
```

- Removed the `Z` prefix because of https://github.com/wordpress-mobile/WordPress-iOS/pull/18660 (even though that hasn't been merged yet, I think we can live with the temporary inconsistency in the interest of closing the loop here)
- This might just be me being pedantic, but I don't like the the use of "magic". There should be no magic in what we do. Pragmatically, it's okay to admit that we don't understand why something happens, _yet_, and that we decided not to invest additional time into it in the interest of tackling more impactful tasks.

To acknowledge the fact that I might be overly pedantic with my "magic avoidance", here's another suggestions just for the `Z` removal. Feel free to apply the one you prefer, no hard feelings 😄 

```suggestion
        // This log magically resolves a deadlock in
        // `DashboardCardTests.testShouldNotShowQuickStartIfDefaultSectionIsSiteMenu`
```

Comment (mokagio) [reply WordPress/WordPressTest/ContextManagerMock.swift]: Related: The previous implementation had a `FIXME` comment.


https://github.com/wordpress-mobile/WordPress-iOS/blob/47fef86327e91cd1decbcab04d32a5c84fdfb344/WordPress/WordPressTest/ContextManagerMock.m#L13

Did you intentionally remove it, or was it "lost in translation"?

I initially thought it would be okay to remove it. But, while writing my comment in favor for it, I realized that it would actually be good to have some semi-automated reminder for this. We have lots of work left to do in the overall Core Data area, so it's possible this will sort itself out, but if it doesn't, there might be something going on underneath that we might want to investigate at some point.

What do you think? An alternative could be to file a reminder task along the lines of "check if the `NSLog` in `ContextManagerMock` is still needed after every Core Data improvement iteration."

**Review by mokagio — APPROVED** [review]
This is looking good.

None of my comments are blocker, but I won't enable auto-merge so you can let me know whether you'd prefer to merge as-is or spend more time on it. 

_If this wasn't a trial, that is if you had the Contributor status, you could hit the merge button yourself, which would make the process I described above less awkward._

---

Oh, just a note to say that I haven't run the tests locally. But I'm going to start CI for this now.

File: WordPress/WordPressTest/ContextManagerMock.swift
Comment (crazytonyli) [reply WordPress/WordPressTest/ContextManagerMock.swift]: Good catch! It's indeed a lost in translation. I'll add it back.

Comment (crazytonyli) [reply WordPress/WordPressTest/ContextManagerMock.swift]: All nitpicks are welcomed and appreciated! I've [taken the wording](https://github.com/wordpress-mobile/WordPress-iOS/pull/18655/commits/b1ec03aa7e6010ae0d50bbfbfcc419ffc9b2a652) in your suggestion. Thanks! 👍

**Comment by mokagio** [comment]
@crazytonyli, `@AliSoftware` I AFK so, in the interest of closing the loop on this one, I think we can merge without his explicit approval. I think you addressed his comments already anyway. And, if there's something we both missed from his suggestions, we can always followup later.

As such, I enabled auto-merge. 👍 

This way, once:

- [x] @crazytonyli resolves merge conflicts
- [x] @mokagio makes maintainer PR to run tests

this will finally land.

**Comment by crazytonyli** [comment]
Thanks @mokagio. I've resolved the conflicts. 🎉

**Pull request merged**
